app/assets/javascripts/listings/components/listingContainer.js.coffee

```diff
@@ -25,14 +25,21 @@ angular.module('dahlia.components')
       @lotteryResultsListings = ListingService.lotteryResultsListings
 
       @isOwnershipListing = (listing) ->
-        @listing.Tenure == 'New sale' || @listing.Tenure == 'Resale'
+        listing.Tenure == 'New sale' || listing.Tenure == 'Resale'
 
-      @isOpenMatchListing = (listing) ->
-        @openMatchListings.indexOf(listing) > -1
+      @hasOwnAndRentFavorited = (listings) ->
+        ownershipFavorites = (@isOwnershipListing listing for listing in @filterByFavorites listings)
+        (_.some ownershipFavorites) && !(_.every ownershipFavorites)
+
+      @filterByFavorites = (listings) ->
```

Comment:
open to better naming suggestions here! another option is to combine this with the function above, but I thought splitting it up made the tests clearer.